### PR TITLE
SEAL-2340: SPIKE: Upgrade Flagship to .NET 4.7.2

### PR DESCRIPTION
**Pull request by inrulejones**

### CODE REVIEW

**Comment by jelster** [comment]
Although the framework versions were bumped in the various CSPROJ files, I'm still seeing all of the various package references and lib refs point to net45 or net461. To resolve, I believe you'd need to essentially re-install the packages into their respective projects. `Get-Package | Upgrade-Package -IgnoreDependencies -Force` is a snippet that's close (check it first!) to what should do the trick from the VS Package Manager console. 

HTH

File: Developer Samples/AddCollectionMembers/packages.config [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="InRule.Common" version="5.4.0" targetFramework="net461" />
-  <package id="InRule.Repository" version="5.4.0" targetFramework="net461" />
-  <package id="InRule.Runtime" version="5.4.0" targetFramework="net461" />
-  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net461" />
-  <package id="Newtonsoft.Json" version="12.0.1" targetFramework="net461" />
+  <package id="InRule.Common" version="5.4.0" targetFramework="net472" />
```
Comment (jelster) [inline Developer Samples/AddCollectionMembers/packages.config:3]: these and the other samples aren't pointing at the latest version of InRule. 
Could that create an issue when package dependency resolution happens where it might still target (via deps) net45/net461 asms?

File: Developer Samples/LibraryLogEnabler/LibLogEnabler/App_Packages/LibLog.4.2/LibLog.cs [hunk 2]
```diff
@@ -40,23 +40,23 @@
 
 using System.Diagnostics.CodeAnalysis;
 
-[assembly: SuppressMessage("Microsoft.Design", "CA1020:AvoidNamespacesWithFewTypes", Scope = "namespace", Target = "TraceLogger.Logging")]
-[assembly: SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Scope = "member", Target = "TraceLogger.Logging.Logger.#Invoke(TraceLogger.Logging.LogLevel,System.Func`1<System.String>,System.Exception,System.Object[])")]
+[assembly: SuppressMessage("Microsoft.Design", "CA1020:AvoidNamespacesWithFewTypes", Scope = "namespace", Target = "LibLogEnabler.Logging")]
```
Comment (jelster) [inline Developer Samples/LibraryLogEnabler/LibLogEnabler/App_Packages/LibLog.4.2/LibLog.cs:43]: Were there customizations made to this file, or was this change applied by the nuget package's install tooling?

File: Developer Samples/LibraryLogEnabler/LibLogEnabler/LoggingEnabler.cs [hunk 3]
```diff
@@ -1,14 +1,13 @@
 ﻿using System;
 using System.Web;
-//LibLog References
-using TraceLogger.Logging;
-using TraceLogger.Logging.LogProviders;
```
Comment (jelster) [inline Developer Samples/LibraryLogEnabler/LibLogEnabler/LoggingEnabler.cs:5]: Was this change required due to the version change, or some other reason?

File: Developer Samples/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice.csproj [hunk 4]
```diff
@@ -39,12 +39,57 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="ActiproSoftware.Docking.Wpf, Version=13.2.592.0, Culture=neutral, PublicKeyToken=36ff2196ab5654b9, processorArchitecture=MSIL">
```
Comment (jelster) [inline Developer Samples/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice.csproj:42]: are these explicit paths/references correct? it still appears to be pointing at net461 for these libs (and an older inrule version at that) - is that due to not having net472 packages available yet?

File: Developer Samples/LibraryLogEnabler/LibLogEnabler/App_Packages/LibLog.4.2/LibLog.cs (hunk 2 above)
Comment (ezInRule) [reply Developer Samples/LibraryLogEnabler/LibLogEnabler/App_Packages/LibLog.4.2/LibLog.cs:43]: nuget upgrade automatically updated everything from TraceLogger to LibLogEnabler

File: Developer Samples/LibraryLogEnabler/LibLogEnabler/LoggingEnabler.cs (hunk 3 above)
Comment (ezInRule) [reply Developer Samples/LibraryLogEnabler/LibLogEnabler/LoggingEnabler.cs:5]: nuget upgrade automatically updated everything from TraceLogger to LibLogEnabler, so trace logger is no more, it seems.

File: Developer Samples/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice.csproj (hunk 4 above)
Comment (ezInRule) [reply Developer Samples/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice/Wpf.ObjectAsStateInvoice.csproj:42]: will have to check, everything was auto nuget upgraded, maybe it doesn't have a new version?  will check...
